### PR TITLE
spec/consensus: canonical vs subjective commit

### PR DESCRIPTION
**Pull request by melekes**
Refs https://github.com/tendermint/tendermint/issues/2769

### CODE REVIEW

File: spec/consensus/consensus.md [hunk 1]
```diff
@@ -337,3 +337,13 @@ where two conflicting reorg-proposals are signed.
 Assuming that the external coordination medium and protocol is robust,
 it follows that forks are less of a concern than [censorship
 attacks](#censorship-attacks).
+
+### Canonical vs subjective commit
+
+We distinguish between "canonical" and "subjective" commits. The latter is what
```
Comment (konnov) [inline spec/consensus/consensus.md:343]:
```suggestion
We distinguish between "canonical" and "subjective" commits. A subjective commit is what
```

File: spec/consensus/consensus.md [hunk 2]
```diff
@@ -337,3 +337,13 @@ where two conflicting reorg-proposals are signed.
...
+each validator sees locally when they decide to commit a block. The former is
```
Comment (konnov) [inline spec/consensus/consensus.md:344]:
```suggestion
each validator sees locally when they decide to commit a block. The canonical commit is
```

File: spec/consensus/consensus.md [hunk 3]
```diff
@@ -337,3 +337,13 @@ where two conflicting reorg-proposals are signed.
...
+what is included by the proposer of the next block in the `LastCommit` field of
+the block. This is what makes it canonical and ensures everyone agrees on it,
```
Comment (konnov) [inline spec/consensus/consensus.md:346]:
```suggestion
the block. This is what makes it canonical and ensures every validator agrees on the canonical commit,
```

File: spec/consensus/consensus.md [hunk 4]
```diff
@@ -337,3 +337,13 @@ where two conflicting reorg-proposals are signed.
...
+even if its different from the first +2/3 votes they saw that caused them to
```
Comment (konnov) [inline spec/consensus/consensus.md:347]:
```suggestion
even if it is different from the +2/3 votes a validator has seen, which caused the validator to
```

File: spec/consensus/consensus.md [hunk 5]
```diff
@@ -337,3 +337,13 @@ where two conflicting reorg-proposals are signed.
...
+commit a block. Each block contains a canonical +2/3 commit for the previous
```
Comment (konnov) [inline spec/consensus/consensus.md:348]:
```suggestion
commit the respective block. Each block contains a canonical +2/3 commit for the previous
```

**Comment by marbar3778** [comment]
@brapse please make sure to squash commits, we should disable it in general for this repo, not sure who has perms to make this change

**Comment by ebuchman** [comment]
> please make sure to squash commits, we should disable it in general for this repo. I'm not sure who has perms to make this change

Linear history now required for merging to master

**Pull request merged**